docs(modelcache): add LocalModelNamespaceCache and inline credential docs (#694)
kserve/website: docs/model-serving/generative-inference/modelcache/localmodel.md

apiVersion: serving.kserve.io/v1alpha1
kind: LocalModelCache
metadata:
  name: meta-llama3-8b-instruct
spec:
  sourceModelUri: "hf://meta-llama/meta-llama-3-8b-instruct"
  modelSize: 10Gi
  nodeGroups:
    - workers
  storage:
    key: hf-credentials
```

#### Using `storage.parameters`

Provide additional storage parameters inline, for example to specify an S3 region:

```yaml title="local-model-cache-s3.yaml"
apiVersion: serving.kserve.io/v1alpha1
kind: LocalModelCache
metadata:
  name: s3-model
spec:
  sourceModelUri: "s3://mybucket/mymodel"
  modelSize: 5Gi
  nodeGroups:
    - workers
  storage:
    key: my-s3-key
    parameters:
      type: s3
      region: us-west-2
```

### Option B: ClusterStorageContainer

You can create a `ClusterStorageContainer` resource with `workloadType: localModelDownloadJob` to define the download container spec and credentials. This is useful when you want to centrally manage the download container image, resource limits, and credentials for all model cache download jobs matching a URI prefix.

#### Create the HF Hub token secret

```yaml title="hf-secret.yaml"
apiVersion: v1
kind: Secret
metadata:
  name: hf-secret

>   namespace: kserve-localmodel-jobs

Reviewer: We should document that the kserve-localmodel-jobs namespace also requires this secret, since it is injected by the controller. Users need to make sure the secret exists in that namespace.

I noticed you mentioned this information here: https://github.com/kserve/website/pull/694/changes#diff-60d8806bbb9517786512e23b775ee021baf3e45a324bfdc49b4c3544e5dfbf47R479

But I think this line is more fit for the information.

Author: added the info just before the secret yaml here

In this guide, you've learned how to:

1. Enable the Local Model Cache feature in KServe
2. Configure credentials for model downloads using inline credentials or `ClusterStorageContainer`

> 3. Create a `LocalModelNodeGroup` to define where models will be cached

Reviewer: ```suggestion
3. Create a `LocalModelNodeGroup` to specify which nodes will store models
```
